libs/frontend/src/lib/features/puzzles/components/play-puzzle.svelte

```diff
@@ -58,28 +60,34 @@
 		puzzle = puzzle;
 	}
 
+	async function patience() {
+		// little timeout has never hurt anyone, also indicates better that something actually happened, system is too fast otherwise
+		return new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
![what](https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExNTZxZDBxdjh0YWI2NHdlN2Fhb2pqcTIya2xqZmUxMmFtbXpmNXBxbSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/ukGm72ZLZvYfS/giphy.gif)

Reply:
I shared the app with someone, and saw them struggling with it, they clicked several times and didn't notice it ran, so instead of clicking once or twice they clicked like 5 times, so I tried to resolve the pain point, 500ms is still relatively fast, and doesn't hurt anyone i imagine :pray:

Reply:
I think an artificial delay is a terrible "solution" to the problem. A better solution would be to make the output of the run more obvious. Maybe a little animation on the test case passed/failed icons when they're updated, or even a little toast at the bottom of the screen giving a summary of the test run (eg. "2/4 tests passed, click here to jump to the failed test case").